File: Solutions.cs/Hash Table/0001. Two Sum.cs
namespace Problem1
{

  /// <summary>
  /// 1. Two Sum
  /// https://leetcode.com/problems/two-sum
  /// 
  /// Difficulty Easy
  /// Acceptance 53.2%
  /// 
  /// Array
  /// Hash Table
  /// </summary>
  public class Solution
  {
    public int[] TwoSum(int[] nums, int target)
    {
      var history = new Dictionary<int, int>(nums.Length);
      for (var i = 0; i < nums.Length; i++)
      {
        if (history.TryGetValue(target - nums[i], out var index))
          return new int[] { index, i };

        history[nums[i]] = i;
      }

      throw new ArgumentException();
    }
  }
}

File: Solutions.java/Hash Table/0001. Two Sum.java
/// <summary>
/// 1. Two Sum
/// https://leetcode.com/problems/two-sum
/// 
/// Difficulty Easy
/// Acceptance 53.2%
/// 
/// Array
/// Hash Table
/// </summary>
class Solution
{
  public int[] twoSum(int[] nums, int target)
  {
    var history = new HashMap<Integer, Integer>(nums.length);
    for (var i = 0; i < nums.length; i++)
    {
      if (history.containsKey(target - nums[i]))
      {
        return new int[] { history.get(target - nums[i]), i };
      }

      history.put(nums[i], i);
    }

    return new int[2];
  }
}
